### PR TITLE
[TACHYON-1428] Documented the tachyon package in preparation for the checkstyle rule

### PR DESCRIPTION
**Pull request by HorizonNet**
https://tachyon.atlassian.net/browse/TACHYON-1428

### CODE REVIEW

File: common/src/main/java/tachyon/Version.java [hunk 1]
```diff
@@ -30,10 +30,17 @@ private Version() {} // prevent instantiation
     VERSION = tachyonConf.get(Constants.TACHYON_VERSION);
   }
 
-  // Relative path to Tachyon target jar
+  /**
+   * The relative path to the Tachyon target jar
```
Comment (yupeng9) [inline common/src/main/java/tachyon/Version.java]: Please add the period at the end.

**Comment by yupeng9** [comment]
Thanks for the doc improvement. I left one minor comment.

**Comment by HorizonNet** [comment]
@yupeng9 Thanks for the comment. I updated the PR accordingly.

File: common/src/main/java/tachyon/Version.java [hunk 2]
```diff
@@ -30,10 +30,17 @@ private Version() {} // prevent instantiation
     VERSION = tachyonConf.get(Constants.TACHYON_VERSION);
   }
 
-  // Relative path to Tachyon target jar
+  /**
+   * The relative path to the Tachyon target jar.
+   */
```
Comment (calvinjia) [inline common/src/main/java/tachyon/Version.java]: Could you use `/** comment */` format?

Comment (HorizonNet) [inline common/src/main/java/tachyon/Version.java]: Thanks for the comment. I updated the PR to address this one.

File: common/src/main/java/tachyon/worker/DataServerMessage.java [hunk 3]
```diff
@@ -329,7 +329,7 @@ private void isSend(boolean isSend) {
    *
    * @param socketChannel The socket channel to receive from
    * @return The number of bytes read, possibly zero, or -1 if the channel has reached end-of-stream
-   * @throws IOException
+   * @throws IOException thrown when a non-Tachyon related exception occurs
```
Comment (gpang) [inline common/src/main/java/tachyon/worker/DataServerMessage.java]: I don't think you need the word "thrown"

File: common/src/main/java/tachyon/worker/DataServerMessage.java [hunk 4]
```diff
@@ -405,7 +405,7 @@ public int recv(SocketChannel socketChannel) throws IOException {
    * Sends this message to the specified socket channel. Make sure this is a send message.
    *
    * @param socketChannel The socket channel to send to
-   * @throws IOException
+   * @throws IOException thrown when a non-Tachyon related exception occurs
```
Comment (gpang) [inline common/src/main/java/tachyon/worker/DataServerMessage.java]: I don't think you need the word "thrown"

File: common/src/main/java/tachyon/util/io/PathUtils.java [hunk 5]
```diff
@@ -134,6 +134,7 @@ public static String getParent(String path) throws InvalidPathException {
    * @param path a path
    * @param prefix a prefix
    * @return whether the given path has the given prefix
+   * @throws InvalidPathException thrown when the path or prefix is invalid
```
Comment (gpang) [inline common/src/main/java/tachyon/util/io/PathUtils.java]: I don't think you need the word "thrown"

File: common/src/main/java/tachyon/util/network/NetworkAddressUtils.java [hunk 6]
```diff
@@ -167,11 +167,21 @@ public int getDefaultPort() {
...
+   */
   public static void assertValidPort(final int port) {
     Preconditions.checkArgument(port < 65536, "Port must be less than 65536");
     Preconditions.checkArgument(port >= 0, "Port must be non-negative");
   }
 
+  /**
+   * Checks if the given port in the address is valid.
+   *
+   * @param address the address with the port to check
```
Comment (gpang) [inline common/src/main/java/tachyon/util/network/NetworkAddressUtils.java]: maybe use {@link InetSocketAddress} in the comment?

**Comment by gpang** [comment]
I left a few comments. Thanks for the improvement!

**Comment by HorizonNet** [comment]
@gpang Thanks for the comments. I updated the PR accordingly.

**Comment by gpang** [comment]
LGTM

**Comment by aaudiber** [comment]
LGTM, thanks for the doc improvement

**Pull request merged**